### ISSUE TITLE
FFMPEG + VMAF - Enable on Windows

### ISSUE THREAD

**Issue by kasivisu82** [Netflix/vmaf#184]
Hi, 

i've trying to enable vmaf on ffmpeg on windows 10. i understand that i need to build vmaf first and then link that to ffmpeg and build ffmpeg again. 

Can someone give me the exact steps on how to do that?

**Comment by alandamore** [comment]
I couldn't do it either, however if you go to https://forum.doom9.org/showthread.php?t=174967&page=2 and click on user "Wolfberry"'s signature, there's an FFMpeg build compiled with VMAF. It works (but sometimes doesn't want to start, some bug I suppose), also need to make sure the log file doesn't already exist, I use it, typical syntax from a batch file:

start ffmpeg.exe -i original.mp4 -i compressed.m4v -lavfi "libvmaf=model_path=vmaf_v0.6.1.pkl:log_path=mylogfilename.log:log_fmt=json:psnr=1:ssim=1:ms_ssim=1" -f null -

**Comment by xcasxcursex** [comment]
Build using this excellent script: https://github.com/m-ab-s/media-autobuild_suite


TL;DR instructions:
Download zip from that repo.
Extract zip
Run batch file
Follow instructions on screen
Choose the non-free license
Choose 'yes' to select which ffmpeg parameters to use, edit the file when prompted (path is on-screen), make sure `--enable-libvmaf` is not commented out
Don't build all the extra stuff you don't need, it will take compilation time from circa 30 minutes to circa 3 hours

HTH. Wolfberry disappeared but I think you'll be pleasantly surprised at how simple this script makes things.

**Comment by StormBytePP** [comment]
Apologies for the "necro-bump" on this old thread, but I believe this information is finally relevant for those looking for a modern, native Windows workflow without MSYS2/MinGW.

For anyone struggling with the original FFmpeg Autotools build system in Windows, there is a [GStreamer-maintained FFmpeg port](https://gitlab.freedesktop.org/gstreamer/meson-ports/ffmpeg.git) that uses **Meson**. It builds perfectly with MSVC and, from my own testing, it is a game changer for Windows native development.

To bridge the gap with VMAF, I've submitted **PR #1477** which adds native MSVC support (including SIMD optimizations) to `libvmaf`. Combining that Meson-based FFmpeg port with this native VMAF build finally allows a pure Visual Studio environment for FFmpeg+VMAF.

No need to reopen this issue, as this is also being tracked in the more recent #1097. Hope this helps anyone landing here from a search engine!
